src/lib.rs

```diff
@@ -182,10 +182,125 @@ pub type CommandInfo = argh_shared::CommandInfo<'static>;
 pub trait FromArgs: Sized {
     /// Construct the type from an input set of arguments.
     ///
-    /// The first argument `command_name` is the identifier for the current
-    /// command, treating each segment as space-separated. This is to be
-    /// used in the output of `--help`, `--version`, and similar flags.
+    /// The first argument `command_name` is the identifier for the current command, treating each
+    /// segment as a separate item. This is used by in the output of `--help` in order to print out
```

Comment:
I fleshed it out some more. Turns out users really should only ever pass in `&[argv[0]]` for  this. The other items in the `command_name` really are only used by subcommands for help strings. It might make sense to someday instead change the API such that end users are calling some new method, like `TopLevelCommand::from_args(args: &[&str])`, which internally could use `FromArgs::from_args(command_name: &[&str], args: &[&str])`, but I think we can consider this down the road.